pkg/controller/node/node_controller.go

```diff
@@ -1288,14 +1289,25 @@ func (ctrl *Controller) updateCandidateNode(mosc *mcfgv1.MachineOSConfig, mosb *
 		}
 
 		lns := ctrlcommon.NewLayeredNodeState(oldNode)
+		desiredConfig := ""
+		desiredImage := ""
 		if !layered {
 			lns.SetDesiredStateFromPool(pool)
+			// If pool is not layered, the desired image annotation is removed (see the `delete`
+			// call in `SetDesiredStateFromPool`), so only the desired config version must be set.
+			desiredConfig = lns.GetDesiredAnnotationsFromMachineConfigPool(pool)
 		} else {
 			lns.SetDesiredStateFromMachineOSConfig(mosc, mosb)
+			desiredConfig, desiredImage = lns.GetDesiredAnnotationsFromMachineOSConfig(mosc, mosb)
 		}
 
-		// Set the desired state to match the pool.
+		// Populate the desired config version and image annotations in the node's MCN
+		err = upgrademonitor.UpdateMachineConfigNodeSpecDesiredAnnotations(ctrl.fgHandler, ctrl.client, nodeName, desiredConfig, desiredImage)
```

Comment:
Should we do this on every sync? Or just when the `reflect.DeepEqual(newData, oldData)` is different?

(I guess the flip side of that question is when does the status get initialized, and how often we'd need to update it)